src/libraries/System.Linq.Queryable/tests/ChunkTests.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class ChunkTests : EnumerableBasedTests
+    {
+
+        [Fact]
+        public void ThrowsOnNullSource()
+        {
+            IQueryable<int> source = null;
+            AssertExtensions.Throws<ArgumentNullException>("source", () => source.Chunk(5));
+        }
+
+        [Fact]
+        public void Chunk()
+        {
+            var count = new[] { 0, 1, 2 }.AsQueryable().Chunk(2).Count();
+            Assert.Equal(2, count);
```

Comment:
Perhaps also validate chunk contents?

Reply:
Will do! 

Is there anything particular  I should be aware of when testing `IQueryable<T>` compared to `IEnumerable<T>`.

Reply:
In most cases validating that the right method has been invoked when run as IEnumerable should suffice, and you probably won't need extensive functional tests. [Other IQueryable tests](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq.Queryable/tests/TakeTests.cs) should provide a good guide in that regard.